src/core_modules/capture-core/components/TemplateSelector/hooks/useProgramStageTemplates.ts

```diff
@@ -10,26 +16,24 @@ export const useProgramStageTemplates = (programId: string) => {
             () => ({
                 templates: {
                     resource: 'programStageWorkingLists',
-                    params: ({ variables }) => ({
-                        filter: `program.id:eq:${variables.programId}`,
+                    params: {
+                        filter: `program.id:eq:${programId}`,
                         fields: 'id,displayName,access,sortOrder',
-                    }),
+                    },
                 },
             }),
-            [],
+            [programId],
         ),
         { lazy: true },
     );
 
     useEffect(() => {
-        supportsStoreProgramStageWorkingList && refetch({ variables: { programId } });
```

Comment:
I think we should keep the same inputs to the refetch function. Currently, this breaks. Same goes for `useTEITemplates`